code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -127,8 +131,13 @@
 	if(status && clumsy_check(user) && prob(50))
 		user.simple_message("<span class='warning'>You grab the [src] on the wrong side.</span>",
 			"<span class='danger'>The [name] blasts you with its power!</span>")
-		user.Knockdown(stunforce)
-		user.Stun(stunforce)
+		if(istype(src, /obj/item/weapon/melee/baton/harm))
```

Comment:
this isn't OOP!

there are 2 better ways to do this
1. Just have a var it checks, and set the var to something different on your new harmbaton; e.g.: (`if(hittype == BATON_AGONY)`...
2. Create a `baton_hit(mob/victim)` proc and call that - have normal behavior in the baton and then override it in your harm baton